client/src/client.rs

```diff
@@ -556,31 +578,31 @@ impl<B, E, Block, RA> Client<B, E, Block, RA> where
 			self.backend.blockchain().expect_block_number_from_id(&BlockId::Hash(max))?,
 		);
 
-		// FIXME: remove this in https://github.com/paritytech/substrate/pull/3201
-		let config_range = ChangesTrieConfigurationRange {
-			config: &config,
-			zero: Zero::zero(),
-			end: None,
-		};
-
 		// fetch key changes proof
-		let first_number = self.backend.blockchain()
-			.expect_block_number_from_id(&BlockId::Hash(first))?;
-		let last_number = self.backend.blockchain()
-			.expect_block_number_from_id(&BlockId::Hash(last))?;
-		let key_changes_proof = key_changes_proof::<Blake2Hasher, _>(
-			config_range,
-			&recording_storage,
-			first_number,
-			&ChangesTrieAnchorBlockId {
-				hash: convert_hash(&last),
-				number: last_number,
-			},
-			max_number,
-			storage_key.as_ref().map(|sk| sk.0.as_slice()),
-			&key.0,
-		)
-		.map_err(|err| sp_blockchain::Error::from(sp_blockchain::Error::ChangesTrieAccessFailed(err)))?;
+		let mut proof = Vec::new();
+		for (config_zero, config_end, config) in configs {
+			let last_number = self.backend.blockchain()
+				.expect_block_number_from_id(&BlockId::Hash(last))?;
+			let config_range = ChangesTrieConfigurationRange {
+				config: &config,
+				zero: config_zero,
+				end: config_end.map(|(config_end_number, _)| config_end_number),
+			};
+			let proof_range = key_changes_proof::<Blake2Hasher, _>(
+				config_range,
+				&recording_storage,
+				first_number,
+				&ChangesTrieAnchorBlockId {
+					hash: convert_hash(&last),
+					number: last_number,
+				},
+				max_number,
+				storage_key.as_ref().map(|x| &x.0[..]),
+				&key.0,
+			)
+			.map_err(|err| sp_blockchain::Error::ChangesTrieAccessFailed(err))?;
+			proof.extend(proof_range);
```

Comment:
Is extend in the right order ?

Reply:
It doesn't need any ordering guarantees. Proof is simply `All touched nodes of all changes tries`